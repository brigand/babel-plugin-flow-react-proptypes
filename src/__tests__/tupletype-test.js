const babel = require('babel-core');
const content = `
var React = require('react');

type T = {
  an_optional_string?: string,
  tupletype: [string, Object]
}

export default function Foo(props: T) {
    <div />
}
`;

it('tupletype', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['env', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
