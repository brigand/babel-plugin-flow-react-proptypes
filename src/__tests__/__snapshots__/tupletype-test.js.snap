// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tupletype 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = Foo;

var React = require('react');

function Foo(props) {
  React.createElement('div', null);
}
Foo.propTypes = {
  an_optional_string: require('prop-types').string,
  tupletype: require('prop-types').arrayOf(function (props, propName, componentName) {
    if (!Object.prototype.hasOwnProperty.call(props, propName)) {
      throw new Error('Prop \`' + propName + '\` has type \\\\'any\\\\', but was not provided to \`' + componentName + '\`. Pass undefined or any other value.');
    }
  }).isRequired
};"
`;
